docs/development/pigs/pig-013.rst

```diff
@@ -0,0 +1,254 @@
+Gammapy modules, which means that import of some Gammapy sub-package such as
+e.g. ``import gammapy.detect`` will often fail should that dependency not be
+installed (which usually doesn't happen, unless users do uncommon things like
+e.g. ``pip uninstall scipy`` after ``pip install gammapy``, that would lead to a
+non-working Gammapy installation). Initially we only had Numpy and Astropy as
+required dependencies, by now we have added Scipy (in `GH 1863`_), PyYAML, click
+and regions as required dependencies.
+
+In Gammapy, we are careful and conservative concerning dependencies. E.g. we
+removed the use of `scikit-image`_ (in  `GH 1658`_) because really
+``scipy.ndimage`` contains everything we need, and also e.g. ``h5py`` and
+``pandas`` and ``ROOT`` and a few other optional dependencies that were only
+used in one or two functions in prototype code were removed (see e.g. `GH
+1586`_).
+
+In the following sections, we summarise what we have now (as of June 2019, in
+Gammapy v0.12), and changes we propose to be made immediately after the Gammapy
+v0.13 release in late July 2019, which will ship to users in Gammapy v0.14 in
+September 2019.
+
+As mentioned already in :ref:`pig-013`, there is always a benefit and a cost to
+supporting old versions. In a few sections below we highlight some of the
+benefits and costs for different dependencies specifically, but generally we
+feel that the benefits of the changes proposed here are significant (much
+reduced testing and maintenance effort, some new features become available),
+while the benefits of keeping support for older versions is very small, since
+most Gammapy users use the recommended conda environment, which already ships
+very modern and recent stack and is mostly unchanged. Especially the
+``astropy-regions`` and ``astropy-healpix`` packages are still under active
+development, and supporting old versions from 2018 or 2017 would be painful, and
+if we decide to require very recent versions of those packages (from summer
+2019), the motivation to support very old versions of other dependencies (most
+notably Astropy, but also e.g. Numpy or Scipy or Matplotlib) is very small.
+
+Required dependencies
+=====================
+
+We propose to update the Gammapy required dependencies as shown in the following
+table (the release dates for the packages are shown in parentheses are were
+obtained from https://pypi.org/).
+
+===============  ===============  ===============
+Dependency       Gammapy 0.12     Gammapy 0.14
+===============  ===============  ===============
+Python           3.5  (Sep 2015)  3.6  (Dec 2016)
+Numpy            1.10 (May 2016)  1.16 (Jan 2019)
+Scipy            0.15 (Jan 2015)  1.2  (Dec 2018)
+Astropy          2.0  (Jul 2017)  3.2  (Jun 2019)
+regions          0.3  (Sep 2018)  0.5  (Jul 2019)
+pyyaml           unclear          3.13 (Jul 2018)
+click            unclear          7.0  (Sep 2018)
+===============  ===============  ===============
+
```

Comment:
Maybe add  a short description of the required dependencies and what they are used for in Gammapy. Also outline the reason for the choice and why they are "required". The reasoning for choosing Astropy, Numpy and Scipy is easy, but for pyyaml and click it's less clear, because there are alternatives...

Reply:
done